server/src/main/java/org/elasticsearch/index/analysis/AnalysisRegistry.java

```diff
@@ -726,18 +726,23 @@ private static NamedAnalyzer produceAnalyzer(
         NamedAnalyzer analyzer;
         if (analyzerF instanceof NamedAnalyzer) {
             // if we got a named analyzer back, use it...
-            analyzer = (NamedAnalyzer) analyzerF;
-            if (overridePositionIncrementGap >= 0 && analyzer.getPositionIncrementGap(analyzer.name()) != overridePositionIncrementGap) {
-                // unless the positionIncrementGap needs to be overridden
-                analyzer = new NamedAnalyzer(analyzer, overridePositionIncrementGap);
-            }
+            analyzer = overrideAnalyzer(analyzerF, overridePositionIncrementGap);
         } else {
             analyzer = new NamedAnalyzer(name, analyzerFactory.scope(), analyzerF, overridePositionIncrementGap);
         }
         checkVersions(analyzer);
         return analyzer;
     }
 
+    private static NamedAnalyzer overrideAnalyzer(Analyzer analyzerF, int overridePositionIncrementGap) {
```

Comment:
We can take a NamedAnalyzer already, given we cast outside of this method? 

Can we also call it overridePositionIncrementGap which seems more specific to what the method does?

Reply:
It was already set to NameAnalyzer the argument in the method. Maybe it needed to be refreshed.
Changed the name also to overridePositionIncrementGap.